core/engine/src/builtins/temporal/plain_year_month/mod.rs

```diff
@@ -338,3 +465,84 @@ pub(crate) fn create_temporal_year_month(
     // 9. Return object.
     Ok(obj.into())
 }
+
+// 9.5.9 AddDurationToOrSubtractDurationFromPlainYearMonth ( operation, yearMonth, temporalDurationLike, options )
+fn add_or_subtract_duration(
+    is_addition: bool,
+    this: &JsValue,
+    duration_like: &JsValue,
+    options: &JsObject,
+    context: &mut Context,
+) -> JsResult<JsValue> {
+    let duration: Duration = if duration_like.is_object() {
+        to_temporal_duration(duration_like, context)?
+    } else if duration_like.is_string() {
+        Duration::from_str(
+            duration_like
+                .as_string()
+                .expect("Value passed not a string")
+                .to_std_string_escaped()
+                .as_str(),
+        )
+        .expect("Unable to parse Duration from string")
+    } else {
+        return Err(JsNativeError::typ()
+            .with_message("cannot handler string durations yet.")
+            .into());
+    };
+
+    let overflow =
+        get_option(options, js_str!("overflow"), context)?.unwrap_or(ArithmeticOverflow::Constrain);
+
+    let year_month = this
+        .as_object()
+        .and_then(JsObject::downcast_ref::<PlainYearMonth>)
+        .ok_or_else(|| {
+            JsNativeError::typ().with_message("this value must be a PlainYearMonth object.")
+        })?;
+
+    let inner = &year_month.inner;
+    let year_month_result = if is_addition {
+        inner
+            .add_duration(&duration, overflow)
+            .expect("Error adding duration to year month")
```

Comment:
issue: The expect here and on line 512 would cause a panic instead of returning any error that happened during addition and subtraction. They should be removed and the error returned.